src/plugins/intel_gpu/src/kernel_selector/kernels/resample/resample_kernel_onnx.cpp

```diff
@@ -0,0 +1,173 @@
+    k.EnableReampleType(ResampleType::LINEAR_ONNX);
+    k.EnableSubGroup();
+    k.EnableSubGroupShort();
+    return k;
+}
+
+ResampleKernelBase::DispatchData ResampleKernelOnnx::SetDefault(const kernel_selector::resample_params &arg) const {
+    DispatchData dispatchData;
+    std::vector<std::vector<Tensor::DataChannelName>> dims_by_gws;
+
+    const auto& out = arg.outputs[0];
+
+    auto opt_x_block_size = GetOptimalBlockSize(arg);
+    if (out.X().v > 32 && opt_x_block_size == 1) {
+        opt_x_block_size = GetOptimalDivisor(out.X().v, 32);
+    }
+
+    dispatchData.gws[0] = CeilDiv(out.X().v, opt_x_block_size) * out.Y().v * out.Z().v;
+    dispatchData.gws[1] = Align(out.Feature().v, sub_group_size);
+    dispatchData.gws[2] = arg.outputs[0].Batch().v;
+
+    dispatchData.lws[0] = 1;
+    dispatchData.lws[1] = sub_group_size;
+    dispatchData.lws[2] = 1;
+
+    if (arg.outputs[0].GetLayout() == DataLayout::bs_fs_zyx_bsv32_fsv16 ||
+        arg.outputs[0].GetLayout() == DataLayout::bs_fs_zyx_bsv32_fsv32) {
+        dispatchData.lws[2] = GetOptimalDivisor(dispatchData.gws[2]);
+    }
+
+    return dispatchData;
+}
+
+KernelsPriority ResampleKernelOnnx::GetKernelsPriority(const Params& /*params*/, const optional_params& /*options*/) const {
+    return FORCE_PRIORITY_4;
+}
+
+bool ResampleKernelOnnx::Validate(const Params& p, const optional_params& o) const {
+    const resample_params& params = static_cast<const resample_params&>(p);
+    if (params.resampleType != ResampleType::LINEAR_ONNX)
+        return false;
+
+    if (!Parent::Validate(p, o))
+        return false;
+
+    if (p.GetType() != KernelType::RESAMPLE || o.GetType() != KernelType::RESAMPLE)
+        return false;
+
+    if (params.inputs.empty())
+        return false;
+
+    const auto& input = params.inputs[0];
+    const auto & output = params.outputs[0];
+    if ( input.Batch().v != output.Batch().v || input.Feature().v != output.Feature().v)
+        return false;
+
+    return true;
+}
+
+static bool IsThreeSpatialResample(const resample_params &params) {
+    const auto& input = params.inputs[0];
+    const auto & output = params.outputs[0];
+
+    if (input.Dimentions() == 5 && input.Z().v != output.Z().v)
+        return true;
+
+    return false;
+}
+
+JitConstants ResampleKernelOnnx::GetJitConstants(const resample_params &params) const {
+    auto jit = Parent::GetJitConstants(params);
+
+    auto opt_x_block_size = GetOptimalBlockSize(params);
+    if (params.outputs[0].X().v > 32 && opt_x_block_size == 1) {
+        opt_x_block_size = GetOptimalDivisor(params.outputs[0].X().v, 32);
+    }
+
+    jit.AddConstant(MakeJitConstant("OUTPUT_X_BLOCK_SIZE", opt_x_block_size));
+    jit.AddConstant(MakeJitConstant("X_BLOCKS", CeilDiv(params.outputs[0].X().v, opt_x_block_size)));
+    jit.AddConstant(MakeJitConstant("SUB_GROUP_SIZE", sub_group_size));
+
+    size_t vec_size = 0;
+    if (params.inputs[0].GetLayout() == DataLayout::fs_b_yx_fsv32) {
```

Comment:
`fs_b_yx_fsv32` is not listed in supported format key. So I think the processing for it can be removed from here.

Reply:
Thanks for the comment, probably it is a reason why there is performance degradation.
I missed  4d formats. This new kernel supports  ONNX interpolation for 4d and 5d blocked formats.